receiver/hostmetricsreceiver/internal/scraper/cpuscraper/internal/metadata/generated_metrics.go

```diff
@@ -14,84 +14,134 @@
 
 // Code generated by mdatagen. DO NOT EDIT.
 
+// This file contains an experimental metrics builder generated by "mdatagen --experimental-gen metadata.yaml" command.
+
 package metadata
 
 import (
+	"time"
+
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/model/pdata"
 )
 
 // Type is the component type name.
 const Type config.Type = "cpu"
 
-// MetricIntf is an interface to generically interact with generated metric.
-type MetricIntf interface {
-	Name() string
-	New() pdata.Metric
-	Init(metric pdata.Metric)
+// MetricConfig provides common settings for a particular metric.
+type MetricConfig struct {
+	Enabled bool `mapstructure:"enabled"`
 }
 
-// Intentionally not exposing this so that it is opaque and can change freely.
-type metricImpl struct {
-	name     string
-	initFunc func(pdata.Metric)
+// MetricsConfig provides settings for cpu metrics.
+type MetricsConfig struct {
+	SystemCPUTime MetricConfig `mapstructure:"system.cpu.time"`
 }
 
-// Name returns the metric name.
-func (m *metricImpl) Name() string {
-	return m.name
+func CreateDefaultMetricsConfig() MetricsConfig {
+	return MetricsConfig{
+		SystemCPUTime: MetricConfig{
+			Enabled: true,
+		},
+	}
 }
 
-// New creates a metric object preinitialized.
-func (m *metricImpl) New() pdata.Metric {
-	metric := pdata.NewMetric()
-	m.Init(metric)
-	return metric
+type metrics struct {
+	SystemCPUTime pdata.Metric
 }
 
-// Init initializes the provided metric object.
-func (m *metricImpl) Init(metric pdata.Metric) {
-	m.initFunc(metric)
+// MetricsBuilder provides an interface for scrapers to report metrics while taking care of all the transformations
+// required to produce metric representation defined in metadata and user configuration.
+type MetricsBuilder struct {
+	config                    MetricsConfig
+	startTime                 pdata.Timestamp
+	initialDataPointsCapacity int
+	metrics                   metrics
 }
 
-type metricStruct struct {
-	SystemCPUTime MetricIntf
+// MetricBuilderOption applies changes to default metrics builder.
+type MetricBuilderOption func(*MetricsBuilder)
```

Comment:
you want the "func(*MetricsBuilder)" to work on something internal (private) so people cannot implement an option.

Reply:
made metricBuilderOption private